src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -1141,8 +1141,10 @@ ExecuteModifyTasks(List *taskList, bool expectResults, ParamListInfo paramListIn
 											 &currentAffectedTupleCount);
 			}
 
-			/* should have rolled back on error */
-			Assert(queryOK);
+			if (!queryOK)
+			{
```

Comment:
Could use a small comment saying that we error out if a query fails, but return false if the connection fails.